### PR TITLE
DateField: fix locales in MM/DD/YYYY format, upgraded package, added unit tests

### PR DESCRIPTION
**Pull request by AlbertCarreras**
#### What changed?

DateField: fix locales in MM/DD/YYYY format, upgraded package, added unit tests

Assessment: https://coda.io/d/_d2LeXkQ1kVX/DateField_sukKr#_luoyc

BEFORE
Screenshot by Dropbox Capture

AFTER
Screenshot by Dropbox Capture

### CODE REVIEW

File: packages/gestalt-datepicker/src/DateField/InternalDateField.js [hunk 1]
```diff
@@ -263,6 +285,17 @@ function InternalDateField({
     }
```
Comment (rlingineni) [inline packages/gestalt-datepicker/src/DateField/InternalDateField.js:285]: I think we should split out into separate functions for readability

```
let translations = getTranslationsFromMUIJS(code);
 if(!locale) {
  translations = getLocalTranslations(code)
 }
```

**Review by rlingineni — APPROVED** [review]
Works, some comments on possible re-ordering.

Can you also add a test for a MUI JS locale and non MUI-JS, and incorrect locale code

File: packages/gestalt-datepicker/src/DateField/InternalDateField.js (hunk 1 above)
Comment (AlbertCarreras) [reply packages/gestalt-datepicker/src/DateField/InternalDateField.js:285]: DONE

**Comment by AlbertCarreras** [comment]
> Works, some comments on possible re-ordering.
> 
> Can you also add a test for a MUI JS locale and non MUI-JS, and incorrect locale code

speaking of tests ha

**Pull request merged**